docs/source/th/optimization/coreml.md

```diff
@@ -0,0 +1,158 @@
+snapshot_download(repo_id, allow_patterns=f"{variant}/*", local_dir=model_path, local_dir_use_symlinks=False)
+print(f"Model downloaded at {model_path}")
+```
+
+### การทำนาย[[python-inference]]
+
+เมื่อคุณดาวน์โหลดสแนปช็อตของโมเดลแล้ว คุณสามารถทดสอบได้โดยใช้สคริปต์ Python ของ Apple
+
+```shell
+python -m python_coreml_stable_diffusion.pipeline --prompt "a photo of an astronaut riding a horse on mars" -i models/coreml-stable-diffusion-v1-4_original_packages -o </path/to/output/image> --compute-unit CPU_AND_GPU --seed 93
+```
+
+`<output-mlpackages-directory>` ควรชี้ไปที่จุดตรวจสอบที่คุณดาวน์โหลดในขั้นตอนก่อนหน้า และ `--compute-unit` ระบุฮาร์ดแวร์ที่คุณต้องการให้ทำนาย ต้องเป็นหนึ่งในตัวเลือกเหล่านี้: `ALL`, `CPU_AND_GPU`, `CPU_ONLY`, `CPU_AND_NE` คุณยังสามารถระบุเส้นทางเอาท์พุททางเลือกได้ตามต้องการ และเลือกเมล็ดพันธุ์สำหรับการทำให้เล่นซ้ำได้
+
+สคริปต์การทำนายถูกแปลว่าคุณกำลังใช้เวอร์ชันเดิมของโมเดล Stable Diffusion คือ `CompVis/stable-diffusion-v1-4` หากคุณใช้โมเดลอื่น เคาะ *ต้อง* ระบุ id Hub ของโมเดลในบรรทัดคำสั่งการทำนาย โดยใช้ตัวเลือก `--model-version` นี้ทำงานสำหรับโมเดลที่ได้รับการสนับสนุนและโมเดลที่ปรับปรุงเอง
+
+ตัวอย่างเช่น หากคุณต้องการใช้ [`runwayml/stable-diffusion-v1-5`](https://huggingface.co/runwayml/stable-diffusion-v1-5):
+
+```shell
+python -m python_coreml_stable_diffusion.pipeline --prompt "a photo of an astronaut riding a horse on mars" --compute-unit ALL -o output --seed 93 -i models/coreml-stable-diffusion-v1-5_original_packages --model-version runwayml/stable-diffusion-v1-5
+```
+
+## Core ML inference ใน Swift
+
+การรันการทำนายใน Swift จะเร็วกว่าน้อยเนื่องจากโมเดลได้ถูกคอมไพล์ไปแล้วในรูปแบบ `mlmodelc` ซึ่งสังเหตุได้ในการเริ่มต้นแอปเมื่อโมเดลถูกโหลด แต่ไม่ควรสังเกตเห็นถึงเมื่อคุณรันการสร้างหลายรุ่นหลังจากนั้น
+
+### การดาวน์โหลด
+
+เพื่อรันการทำนายใน Swift บน Mac ของคุณ คุณต้องใช้หนึ่งในตัวแปรที่ถูกคอมไพล์เช็คพอยท์เวอร์ชัน `compiled` เราขอแนะนำให้คุณดาวน์โหลดไฟล์เหล่านี้ไปยังเครื่องโดยใช้โค้ด Python ที่คล้ายกับตัวอย่างก่อนหน้า แต่ใช้ตัวแปร `compiled` หนึ่งในตัวเลือก:
+
+```Python
+from huggingface_hub import snapshot_download
+from pathlib import Path
+
+repo_id = "apple/coreml-stable-diffusion-v1-4"
+variant = "original/compiled"
+
+model_path = Path("./models") / (repo_id.split("/")[-1] + "_" + variant.replace("/", "_"))
+snapshot_download(repo_id, allow_patterns=f"{variant}/*", local_dir=model_path, local_dir_use_symlinks=False)
+print(f"Model downloaded at {model_path}")
+```
+
+### การทำนาย[[swift-inference]]
+
+เพื่อรันการทำนาย โปรดคล๊อนเรพอ Apple's repo:
+
+```bash
+git clone https://github.com/apple/ml-stable-diffusion
+cd ml-stable-diffusion
+```
+
+แล้วใช้เครื่องมือของ Apple คือ [Swift Package Manager](https://www.swift.org/package-manager/#):
+
+```bash
+swift run StableDiffusionSample --resource-path models/coreml-stable-diffusion-v1-4_original_compiled --compute-units all "a photo of an astronaut riding a horse on mars"
+```
+
+คุณต้องระบุเส้นทางไปยังโมเดล Stable Diffusion ที่คุณดาวน์โหลดในขั้นตอนก่อนหน้า
```

Comment:
I can't really be sure, but it looks like we may be missing the last two sections here ([Supported Diffusers Features](https://huggingface.co/docs/diffusers/main/en/optimization/coreml#supported-diffusers-features) and Native Diffusers Swift app).